runtime/commonMain/src/kotlinx/serialization/Shorthands.kt

```diff
@@ -26,12 +26,15 @@ public fun SerialDescriptor.elementDescriptors(): List<SerialDescriptor> {
 }
 
 /**
- * Creates a [List] out of a serial names of [this] descriptor's elements [SerialDescriptor.getElementDescriptor].
- *
- * Size of a list is equal to [SerialDescriptor.elementsCount].
+ * Returns a [Set] out of all serial names of serial descriptor [elements][SerialDescriptor.getElementDescriptor].
  */
-public fun SerialDescriptor.elementNames(): List<String> {
-    return List(elementsCount) { getElementName(it) }
+public fun SerialDescriptor.elementNames(): Set<String> {
```

Comment:
I may be wrong, but for some usages, element order might be crucial – that's why there was a `List`